k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -123,6 +141,23 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
               }
             })
             val watchStream = Json.readStream[W](rsp.reader, Api.BufSize)
+              // it would be nice to not have to flat map over the watch stream, but this is
+              // necessary to handle the K8s bug where erorrs have the incorrect status code.
+              // hopefully this hack can eventually be removed...
+              .flatMap {
+                // special case to handle Kubernetes bug where "too old resource version" errors
+                // are returned with status code 200 rather than status code 410.
+                // see https://github.com/kubernetes/kubernetes/issues/35068 for details.
+                case e: Watch.Error[O] if e.status.message.exists(_.startsWith("too old resource version")) =>
+                  log.warning(
```

Comment:
Is it true that this k8s bug has no adverse effects now that we're detecting "too old" events with status code 200?  If so, I don't think we need to log.  (we will continue to log at trace level that a watch was restarted)

Reply:
Hmm. I thought it was worthwhile to encourage users to upgrade their Kubernetes version, so that we can eventually remove this workaround? I suspect it adds at least a minor performance hit.

Reply:
I suspect this is pretty cheap, actually.  The objects have already been parsed, after all.  And we will never be able to assume that all users have upgrade off of the affected K8s version.  The best we can do is eventually drop support for certain k8s versions.